### PR TITLE
Add functions to create conversations within teams

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

* Add functions to create conversations.
* Add a couple of helper functions.

### CODE REVIEW

File: Source/Model/UserClient/UserClient.swift [hunk 1]
```diff
@@ -200,7 +200,7 @@ public class UserClient: ZMManagedObject, UserClientType {
     
     /// Resets the session between the client and the selfClient
     /// Can be called several times without issues
-    public func resetSession() {
+    public func resetSession(in team: Team?) {
```
Comment (daehn) [inline Source/Model/UserClient/UserClient.swift]: Not sure about this, as there will only be one session per user this is actually not needed and we should figure out which conversation it is inside this method.

File: Source/Model/Conversation/ZMConversation.m [hunk 2]
```diff
@@ -1088,14 +1090,61 @@ + (instancetype)conversationWithRemoteID:(NSUUID *)UUID createIfNeeded:(BOOL)cre
     return nil;
 }
 
-+ (instancetype)insertGroupConversationIntoManagedObjectContext:(NSManagedObjectContext *)moc withParticipants:(NSArray *)participants;
++ (instancetype)fetchOrCreateTeamConversationInManagedObjectContext:(NSManagedObjectContext *)moc withParticipant:(ZMUser *)participant team:(Team *)team
```
Comment (mikeger) [inline Source/Model/Conversation/ZMConversation.m]: Should it be `... inTeam:` ?

Comment (mikeger) [inline Source/Model/Conversation/ZMConversation.m]: Ah, you have `fetchOrCreateTeam` in the name, I guess then it's not necessary.

File: Source/Model/User/ZMUser.m [hunk 3]
```diff
@@ -255,9 +255,15 @@ - (NSString *)initials
     return personName.initials ?: @"";
 }
 
-- (ZMConversation *)oneToOneConversation
```
Comment (mikeger) [inline Source/Model/User/ZMUser.m]: I guess this would break the API, should we leave the old method be too?

Comment (mikeger) [inline Source/Model/User/ZMUser.m]: I have no strong opinion about that, the other methods require team parameter now as well. May be we should mark them as nullable parameters in the headers?

Comment (daehn) [inline Source/Model/User/ZMUser.m]: I thought about it, but it might make sense to break the API here to enforce to change it everywhere on the UI.

Comment (daehn) [inline Source/Model/User/ZMUser.m]: Yeah, I will mark them as `nullable` and also remove the `resetSession(in:)`, as it shouldn't be needed there.

Comment (daehn) [inline Source/Model/User/ZMUser.m]: @mikeger updated!

**Pull request merged**